### PR TITLE
Double quotes

### PR DESCRIPTION
**Pull request by tdiam**
Even though double quotes were included in `CHARS_GLOBAL_REGEXP` there was no escape mapping.
This PR resolves this issue:

```js
const tsqlstring = require('tsqlstring')

console.log(tsqlstring.escape('Sup"er'))
// before this PR: 'Supundefineder'
// after this PR: 'Sup\"er'
```

EDIT: Actually, if I understand it correctly, double quotes do not need escaping unless `QUOTED_IDENTIFIER` is `ON`.
@kylefarris how do you think this should be handled?

### CODE REVIEW

**Comment by samjross** [comment]
This fix doesn't work for me personally. I tested it out locally with an mssql server, and `"\""   : '\\"',` leaves me with a backslash in the actual string on the table.

For me, `"\""   : '"',` worked.

**Comment by samjross** [comment]
Seems to me that in mssql, `"` doesn't actually need to be escaped at all, so the real solution is probably to take `\"` out of `CHARS_GLOBAL_REGEXP`, but I'm not confident that I understand the purpose or scope if this library enough to test that. That's what works for me.

**Comment by kylefarris** [comment]
I think I went through this same conundrum when I was originally writing this library. In T-SQL the double-quote itself can be considered a way to protect identifiers but only if the setting is turned on for the server (something this library may not be able to easily detect). In other words, escaping the double-quote could potentially break queries that already have protected identifiers and cause it to totally break.

So, like @samjross said, I think we just need to remove the `\"` from the `CHARS_GLOBAL_REGEXP` (looks like removing that was just an oversight anyway).

So, @tdiam if you just want to make that change and undo the other stuff, that'd probably fix it. I'd be happy to get this PR merged in.

File: lib/SqlString.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ class SqlString {
             "\r"   : '\\r',
             "\v"   : '\\v',
             "\x1a" : '\\Z',
+            "\""   : '\\"',
```
Comment (kylefarris) [inline lib/SqlString.js]:
```suggestion
```

File: test/unit/test-SqlString.js [hunk 2]
```diff
@@ -228,6 +228,11 @@ test('SqlString.escape', {
     assert.equal(SqlString.escape('Super\''), "'Super'''");
   },
 
+  'double quotes get escaped': function() {
+    assert.equal(SqlString.escape('Sup"er'), "'Sup\\\"er'");
+    assert.equal(SqlString.escape('Super"'), "'Super\\\"'");
+  },
+
```
Comment (kylefarris) [inline test/unit/test-SqlString.js]:
```suggestion
  'double quotes do not get escaped': function() {
    assert.equal(SqlString.escape('Sup"er'), `'Sup"er'`);
    assert.equal(SqlString.escape('Super"'), `'Super"'`);
  },


```

**Review by kylefarris — CHANGES_REQUESTED** [review]

**Comment by kylefarris** [comment]
@tdiam I've updated your remote branch with some suggested edits. The only thing you need to do now is remove the double quote item from `CHARS_GLOBAL_REGEXP`.

**Comment by kylefarris** [comment]
@tdiam, actually, it looks like I was able to edit your branch directly somehow? So, I'll go ahead and merge this in and up date the package on NPM.

**Comment by kylefarris** [comment]
Patch added to NPM as v1.0.1.

**Comment by tdiam** [comment]
Yep, I reached the same conclusions. Thanks for the input and for merging @samjross, @kylefarris !

**Comment by samjross** [comment]
awesome, thank you @kylefarris !

**Comment by samjross** [comment]
@kylefarris I can't see a way to open an issue on this repo, but I've run into a problem with new lines, `\n`

**Pull request merged**